### PR TITLE
Add Apptainer documentation

### PR DESCRIPTION
**Pull request by DrDaveD**

### CODE REVIEW

File: docs/worker-node/install-apptainer.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
+title: Install Apptainer
+
+Install Apptainer
+=================
+
+[Apptainer](http://apptainer.org) (formerly known as Singularity, see
+[announcement](https://apptainer.org/news/community-announcement-20211130))
+is a tool that creates
+docker-like process containers but without giving extra privileges to
+unprivileged users.  It is used by grid pilot jobs (which are
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:10]:
```suggestion
unprivileged users.  It is used by pilot jobs (which are
```

Slightly redundant with the parenthetical.

File: docs/worker-node/install-apptainer.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+and processes.  It also supplies a chroot environment in order to run
+user jobs in different operating system images under one Linux kernel.
+
+Apptainer works either by making use of unprivileged user namespaces
+or with a setuid-root assist program. 
+By default it does not install the setuid-root assist program
+and it uses only unprivileged user namespaces.
+Unprivileged user namespaces are available on all OS versions that
+OSG supports, although it is not enabled by default on RHEL 7;
+instructions to enable it are below.
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:22]:
```suggestion
instructions to enable it are [below](#enabling-unprivileged-apptainer).
```

File: docs/worker-node/install-apptainer.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+---------------
+
+As with all OSG software installations, there are some one-time (per host)
+steps to prepare in advance:
+
+- Ensure the host has [a supported operating system](../release/supported_platforms.md)
+- Obtain root access to the host
+- If you intend to install Apptainer locally, then
+  prepare the [required Yum repositories](../common/yum.md). 
+  Note that the apptainer RPM comes from the EPEL yum repository.
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:50]:
```suggestion
  Note that the apptainer RPM comes from the EPEL Yum repository.
```

File: docs/worker-node/install-apptainer.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+jobs from the pilot's files and processes and from other users' files
+and processes.  It also supplies a chroot environment in order to run
+user jobs in different operating system images under one Linux kernel.
+
+Apptainer works either by making use of unprivileged user namespaces
+or with a setuid-root assist program. 
+By default it does not install the setuid-root assist program
+and it uses only unprivileged user namespaces.
+Unprivileged user namespaces are available on all OS versions that
+OSG supports, although it is not enabled by default on RHEL 7;
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:21]:
```suggestion
OSG supports, although it is not enabled by default on EL 7;
```

File: docs/worker-node/install-apptainer.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+instructions to enable it are below.
+The feature is enabled by default on RHEL 8.
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:23]:
```suggestion
The feature is enabled by default on EL 8.
```

File: docs/worker-node/install-apptainer.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+
+There are two sets of instructions on this page:
+
+- [Enabling Unprivileged Apptainer](#enabling-unprivileged-apptainer)
+- [Apptainer via RPM](#apptainer-via-rpm)
+
+OSG VOs all support running apptainer directly from CVMFS when unprivileged
+user namespaces are enabled.
+Unprivileged user namespaces are enabled by default on RHEL 8, and OSG
+recommends that system administrators enable it on RHEL 7 worker nodes.
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:69]:
```suggestion
Unprivileged user namespaces are enabled by default on EL 8, and OSG
recommends that system administrators enable it on EL 7 worker nodes.
```

File: docs/worker-node/install-apptainer.md [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+If docker is being used to run jobs, the following options are 
+recommended to allow unprivileged Apptainer to run (it does not
+need `--privileged` or any added capabilities):
+
+    ::console
+    --security-opt seccomp=unconfined --security-opt systempaths=unconfined
+
+`--security-opt seccomp=unconfined` enables unshare to be called
+(which is needed to create namespaces),
+and `--security-opt systempaths=unconfined` allows /proc to be mounted
```
Comment (brianhlin) [inline docs/worker-node/install-apptainer.md:147]:
```suggestion
and `--security-opt systempaths=unconfined` allows `/proc` to be mounted
```

**Review by brianhlin — APPROVED** [review]
Minor suggestions, otherwise LGTM

Are we at the point where we should add this to the table of contents by adding a line to `mkdocs.yml` and removing the Singularity doc from the table of contents with a note at the top pointing to the upgrading from Singularity section of this doc?

**Comment by brianhlin** [comment]
Link failures unrelated, merging

**Comment by DrDaveD** [comment]
> Are we at the point where we should add this to the table of contents by adding a line to `mkdocs.yml` and removing the Singularity doc from the table of contents with a note at the top pointing to the upgrading from Singularity section of this doc?

I didn't notice this question before.  No I think we should wait until Apptainer is released to update the table of contents.  Also at that time I plan to replace the singularity document with just a bunch of links to the corresponding apptainer document.

**Comment by brianhlin** [comment]
If you want to keep the page that sounds good, otherwise we can do entire page redirects: https://github.com/osg-htc/docs/blob/master/mkdocs.yml#L112-L115

**Comment by DrDaveD** [comment]
I want to redirect individual sections, and the names aren't always the same between the two, so I think a page of manual forwards is better.

**Pull request merged**